src/modm/math/geometry/location_2d.hpp

```diff
@@ -37,39 +36,60 @@ namespace modm
 	class Location2D
 	{
 	public:
-		Location2D();
-
-		Location2D(const Vector<T, 2>& position, const float& orientation);
-
-		Location2D(const T& x, const T& y, const float& orientation);
-
-		inline const Vector<T, 2>&
-		getPosition() const;
-
-		inline const T&
-		getX() const;
-
-		inline const T&
-		getY() const;
-
-		void
-		setPosition(const Vector<T, 2>& point);
-
-		void
-		setPosition(const T& x, const T& y);
+		Vector<T, 2> position;
+		float orientation = 0;
+
+		constexpr Location2D() = default;
+		constexpr Location2D(const Vector<T, 2>& position, const float& orientation)
+			: position(position), orientation(orientation) {}
+
+		[[deprecated("Use 'setPosition({x, y}, orientation)' instead!")]]
+		constexpr Location2D(const T& x, const T& y, const float& orientation)
+			: position(x, y), orientation(orientation) {}
+
+		template<typename U>
+		constexpr Location2D(const Location2D<U> &l) : position(l.position), orientation(l.orientation) {}
+
+		// getters and setters
+		void setPosition(const Vector<T, 2>& position) { this->position = position; }
+
+		[[deprecated("Use 'setPosition({x, y}' instead!")]]
+		void setPosition(T x, T y) { this->position.x() = x; this->position.y() = y; }
+		void setOrientation(const float orientation) { this->orientation = orientation; }
+
+		Vector<T, 2> getPosition() const { return position; }
+		inline float getOrientation() const { return orientation; }
+		T getX() const { return position.x(); }
+		T getY() const { return position.y(); }
+
+		bool operator== (const Location2D &other) const {
+			return (
+				position == other.position and
+				std::abs(orientation - other.orientation) < __FLT_EPSILON__
```

Comment:
Can position be float? Should we handle that case somehow?

Not sure if `__FLT_EPSILON__` is the right thing to do. For numbers between 1.0 and 2.0 it's the distance between adjacent floats. For numbers greater than 2.0 the comparison will do the same thing as `==`.